sklearn/metrics/_pairwise_distances_reduction/__init__.py

```diff
@@ -0,0 +1,101 @@
+#    on the dtype of X and of Y.
+#
+#
+# High-level diagram
+# ------------------
+#
+#    Legend:
+#
+#      A ---⊳ B: A inherits from B
+#      A ---x B: A dispatches on B
+#
+#
+#                               (base dispatcher)
+#                           PairwiseDistancesReduction
+#                                       ∆
+#                                       |
+#                                       |
+#                     +-----------------+-----------------+
+#                     |                                   |
+#               (dispatcher)                        (dispatcher)
+#         PairwiseDistancesArgKmin           PairwiseDistancesRadiusNeighbors
+#               |                                              |
+#               |                                              |
+#               |                                              |
+#               |                  (64bit implem.)             |
+#               |          PairwiseDistancesReduction64        |
+#               |                       ∆                      |
+#               |                       |                      |
+#               |                       |                      |
+#               |     +-----------------+-----------------+    |
+#               |     |                                   |    |
+#               |     |                                   |    |
+#               x     |                                   |    x
+#        PairwiseDistancesArgKmin64       PairwiseDistancesRadiusNeighbors64
+#               |     ∆                                   ∆    |
+#               |     |                                   |    |
+#               x     |                                   |    |
+#     FastEuclideanPairwiseDistancesArgKmin64             |    |
+#                                                         |    |
+#                                                         |    x
+#                                  FastEuclideanPairwiseDistancesRadiusNeighbors64
+#
+#    For instance :class:`PairwiseDistancesArgKmin`, dispatches to
+#    :class:`PairwiseDistancesArgKmin64` if X and Y are both dense NumPy arrays
+#    with a float64 dtype.
+#
+#    In addition, if the metric parameter is set to "euclidean" or "sqeuclidean",
+#    :class:`PairwiseDistancesArgKmin64` further dispatches to
+#    :class:`FastEuclideanPairwiseDistancesArgKmin64` a specialized subclass
+#    to optimally handle the Euclidean distance case using the Generalized Matrix
+#    Multiplication (see the docstring of :class:`GEMMTermComputer64` for details).
+
+
+from ._dispatcher import (
+    PairwiseDistancesReduction,
+    PairwiseDistancesArgKmin,
+    PairwiseDistancesRadiusNeighborhood,
+    _sqeuclidean_row_norms,
+)
+
+__all__ = [
+    "PairwiseDistancesReduction",
+    "PairwiseDistancesArgKmin",
+    "PairwiseDistancesRadiusNeighborhood",
+    "_sqeuclidean_row_norms",
```

Comment:
I think we can define ``sqeuclidean_row_norms`` without the prefix underscore, like the dispacher classes.